app/models/bulkrax/oai_entry.rb

```diff
@@ -62,16 +61,13 @@ def collections_created?
     def find_or_create_collection_ids
       return self.collection_ids if collections_created?
 
-      if sets.blank?
-        c = Collection.where(Bulkrax.system_identifier_field => parser.collection_name).first
+      if sets.blank? || parser.collection_name != 'all'
+        c = Collection.where(Bulkrax.system_identifier_field => importer.unique_collection_identifier(parser.collection_name)).first
+        self.collection_ids << c.id unless c.blank? || self.collection_ids.include?(c.id)
+      else # All - collections should exist for all sets
+        c = Collection.where(Bulkrax.system_identifier_field => importer.unique_collection_identifier(set.content)).first
```

Comment:
This seems like a big change. Before we were adding every collection id in the all case. Now are are only adding one?

Reply:
that's wrong, fixed now.